### PR TITLE
[dev] Update go to 1.16.3

### PR DESCRIPTION
**Pull request by aledbf**
- Update go version to 1.16.3
- Adjust go build flags 

Why? Using pprof to find a leak (https://github.com/gitpod-io/gitpod/issues/3108) I found that it was not possible to use `pprof list`

**before this change**

```
goroutine profile: total 31
1 @ 0x40c6f4 0x46e545 0x19e2265 0x471e61
#	0x46e544	os/signal.signal_recv+0xa4	/home/gitpod/sdk/go1.16/src/runtime/sigqueue.go:168
#	0x19e2264	os/signal.loop+0x24		/home/gitpod/sdk/go1.16/src/os/signal/signal_unix.go:23

1 @ 0x43aec5 0x4068cf 0x40650b 0x1923b13 0x180be8e 0x180bf51 0x471e61
#	0x1923b12	k8s.io/client-go/tools/cache.(*sharedProcessor).run+0x52		/workspace/go/pkg/mod/k8s.io/client-go@v0.20.4/tools/cache/shared_informer.go:628
#	0x180be8d	k8s.io/apimachinery/pkg/util/wait.(*Group).StartWithChannel.func1+0x2d	/workspace/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:56
#	0x180bf50	k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1+0x50		/workspace/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:73
```

**after**

```
goroutine profile: total 162
100 @ 0x43aec5 0x44bfd7 0x16a20e8 0x16a1c57 0x471e01
#	0x16a20e7	github.com/gitpod-io/gitpod/ws-manager/pkg/manager/internal/workpool.(*EventWorkerPool).work+0x447		github.com/gitpod-io/gitpod/ws-manager/pkg/manager/internal/workpool/workpool.go:149
#	0x16a1c56	github.com/gitpod-io/gitpod/ws-manager/pkg/manager/internal/workpool.(*EventWorkerPool).workUntilStopped+0x56	github.com/gitpod-io/gitpod/ws-manager/pkg/manager/internal/workpool/workpool.go:99

3 @ 0x43aec5 0x4068cf 0x40650b 0x16b34d3 0xd8b84e 0xd8b911 0x471e01
#	0x16b34d2	k8s.io/client-go/tools/cache.(*sharedProcessor).run+0x52		k8s.io/client-go@v0.20.4/tools/cache/shared_informer.go:628
#	0xd8b84d	k8s.io/apimachinery/pkg/util/wait.(*Group).StartWithChannel.func1+0x2d	k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:56
#	0xd8b910	k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1+0x50		k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:73

3 @ 0x43aec5 0x4068cf 0x40650b 0x16b7d94 0x471e01
#	0x16b7d93	k8s.io/client-go/tools/cache.(*controller).Run.func1+0x33	k8s.io/client-go@v0.20.4/tools/cache/controller.go:130
```

### CODE REVIEW

File: components/content-service/BUILD.yaml [hunk 1]
```diff
@@ -28,6 +28,7 @@ packages:
       - GOOS=linux
     config:
       packaging: app
+      buildCommand: ["go", "build", "-trimpath", "-ldflags='-buildid= -w -s'"]
```
Comment (csweichel) [inline components/content-service/BUILD.yaml:31]: You can do this centrally for all Go packages in https://github.com/gitpod-io/gitpod/blob/main/WORKSPACE.yaml#L10

This way only `workspacekit` would need some adjustment.

Comment (aledbf) [reply components/content-service/BUILD.yaml]: fixed

**Review by csweichel — APPROVED** [review]
LGTM. Thanks for the update

**Pull request merged**